statefun-kafka-io/src/main/java/org/apache/flink/statefun/sdk/kafka/KafkaIngressBuilder.java

```diff
@@ -168,13 +174,20 @@ private Properties resolveKafkaProperties() {
     }
   }
 
+  private static <T> void overwritePropertyIfNamedConfigIsPresent(
```

Comment:
Suggestion: Since `OptionalConfig` is a package private class, moving this method into the `OptionalConfig` might make it a bit more readable.

For example:
```
ConsumerConfig.GROUP_ID.upadte(properties, consumerGroupId);
```

Either way is good tho!

Reply:
Makes sense. Updated according to your suggestion.